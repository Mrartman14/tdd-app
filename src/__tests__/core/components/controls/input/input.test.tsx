import React from 'react';
import { configure, shallow, ShallowWrapper } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { act } from "react-dom/test-utils";

import { Input, InputProps } from '../../../../../core/components';

configure({
    adapter: new Adapter(),
});

function setupComponent(props: InputProps = {}): ShallowWrapper {
    let component: ShallowWrapper;

    act(() => {
        component = shallow(<Input {...props} />);
    });

    return component!;
}

describe('should input renders well', () => {
    let component: ShallowWrapper;

    afterEach(() => {
        act(() => {
            jest.resetAllMocks();
            component.unmount();
        });
    });

    // TODO: why this is not passed?
    // it('should input can be disabled', () => {
    //     component = setupComponent({ disabled: true });
    //     expect(component.getElement()).toBeDisabled();
    // });

    it('should onChange prop can be called', () => {
        const mockedOnclick = jest.fn();
        component = setupComponent({ onChange: mockedOnclick });

        expect(mockedOnclick).toHaveBeenCalledTimes(0);

        const changeEvent = {
            target: {
                value: 'test value',
            },
        };

        act(() => {
            component.simulate('change', changeEvent);
        });

        expect(mockedOnclick).toHaveBeenCalledTimes(1);
        expect(mockedOnclick).toHaveBeenLastCalledWith(changeEvent);
    });

    it('snapshot', () => {
        component = setupComponent({
            disabled: true,
        });
        expect(component).toMatchSnapshot();
    });
});
