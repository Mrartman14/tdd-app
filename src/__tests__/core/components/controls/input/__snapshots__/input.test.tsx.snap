// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should input renders well snapshot 1`] = `
<input
  disabled={true}
/>
`;
